README.md

```diff
@@ -79,77 +79,87 @@ This example demonstrates how to transcribe a local audio file:
+For MLX models, see [here](https://huggingface.co/argmaxinc/whisperkit-mlx).
+
+If you want to get the recommended model for your device, you can use the following method:
+
+```swift
+print(WhisperKit.recommendedModel())
+```
 
 ### Generating Models
 
 WhisperKit also comes with the supporting repo [`whisperkittools`](https://github.com/argmaxinc/whisperkittools) which lets you create and deploy your own fine tuned versions of Whisper in CoreML format to HuggingFace. Once generated, they can be loaded by simply changing the repo name to the one used to upload the model:
 
 ```swift
-let pipe = try? await WhisperKit(model: "large-v3", modelRepo: "username/your-model-repo")
+let pipe = try await WhisperKit(model: "large-v3", modelRepo: "username/your-model-repo")
 ```
 
-### Swift CLI
-
-The Swift CLI allows for quick testing and debugging outside of an Xcode project. To install it, run the following:
+### Backend Selection
 
-```bash
-git clone https://github.com/argmaxinc/whisperkit.git
-cd whisperkit
-```
+WhisperKit supports both CoreML and MLX backends. By default, it uses CoreML, but you can switch some or all pipeline components to MLX.
+Available pipeline components are:
+- `featureExtractor`, `FeatureExtractor` is used by default, use `MLXFeatureExtractor` to switch to MLX
+- `audioEncoder`, `AudioEncoder` is used by default, use `MLXAudioEncoder` to switch to MLX
+- `textDecoder`, `TextDecoder` is used by default, use `MLXTextDecoder` to switch to MLX
 
-Then, setup the environment and download your desired model.
+Here is an example of how to switch the `featureExtractor` and `audioEncoder` to MLX and keep the `textDecoder` as CoreML:
 
-```bash
-make setup
-make download-model MODEL=large-v3
+```swift
+let pipe = try await WhisperKit(
+  model: "tiny", 
+  mlxModel: "tiny", 
+  featureExtractor: MLXFeatureExtractor(),
+  audioEncoder: MLXAudioEncoder()
+)
 ```
 
-**Note**:
+### Testing
 
-1. This will download only the model specified by `MODEL` (see what's available in our [HuggingFace repo](https://huggingface.co/argmaxinc/whisperkit-coreml), where we use the prefix `openai_whisper-{MODEL}`)
-2. Before running `download-model`, make sure [git-lfs](https://git-lfs.com) is installed
-
-If you would like download all available models to your local folder, use this command instead:
+If you want to run the unit tests locally, first clone the repo:
 
 ```bash
-make download-models
+git clone https://github.com/argmaxinc/whisperkit.git
+cd whisperkit
 ```
 
-You can then run them via the CLI with:
+download the required models:
 
 ```bash
-swift run whisperkit-cli transcribe --model-path "Models/whisperkit-coreml/openai_whisper-large-v3" --audio-path "path/to/your/audio.{wav,mp3,m4a,flac}" 
+make setup
+make download-model MODEL=tiny
+make download-mlx-model MODEL=tiny
 ```
 
-Which should print a transcription of the audio file. If you would like to stream the audio directly from a microphone, use:
+and then run the tests:
```

Comment:
Seems like some useful stuff got deleted in this section based on the diff here.

Reply:
reverted